src/sentry/integrations/gitlab/client.py

```diff
@@ -64,59 +76,40 @@ def get_group(self, group):
         path = GitLabApiClientPath.group.format(group=group)
         return self.get(path)
 
-    def request(self, method, path, data=None, params=None):
-        headers = {"Authorization": f"Bearer {self.token}"}
-        return self._request(
-            method,
-            GitLabApiClientPath.build_api_url(self.base_url, path),
-            headers=headers,
-            data=data,
-            params=params,
-        )
-
 
-class GitLabApiClient(ApiClient):
+class GitLabProxyApiClient(IntegrationProxyClient):
     integration_name = "gitlab"
 
     def __init__(self, installation):
         self.installation = installation
         verify_ssl = self.metadata["verify_ssl"]
         self.is_refreshing_token = False
+        self.new_identity = None
+        self.base_url = self.metadata["base_url"]
         super().__init__(verify_ssl)
```

Comment:
The `IntegrationProxyClient` will require an `org_integration_id` as part of it's `__init__` so that it can add it as part of the proxy details along in the request (https://github.com/getsentry/sentry/blob/master/src/sentry/shared_integrations/client/proxy.py#L86-L100)

The 'SetupClient' probably doesn't have an `org_integration_id` to provide though, so you can do there [what I did for Slack's setup](https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/slack/integration.py#L152-L161) as well, just manually add the auth headers since it'll only run in Control.